webpack/assets/javascripts/react_app/components/ModelsTable/ModelsTableReducer.js

```diff
@@ -0,0 +1,43 @@
+import Immutable from 'seamless-immutable';
+import {
+  HW_MODEL_TABLE_REQUEST,
+  HW_MODEL_TABLE_SUCCESS,
+  HW_MODEL_TABLE_FAILURE,
+} from './ModelsTableConsts';
+import { STATUS } from '../../constants';
+
+const initState = Immutable({
+  page: 1,
+  perPage: 20,
+  total: 0,
+  sortBy: '',
+  sortOrder: '',
+  results: [],
+  error: null,
+  status: STATUS.PENDING,
+  searchQuery: null,
+});
+export default (state = initState, action) => {
+  switch (action.type) {
+    case HW_MODEL_TABLE_REQUEST:
+      return state.set('status', STATUS.PENDING);
+    case HW_MODEL_TABLE_SUCCESS:
+      return Immutable.merge(state, {
+        status: STATUS.RESOLVED,
+        error: null,
+        searchQuery: action.payload.search,
+        results: action.payload.results,
+        page: action.payload.page,
+        perPage: action.payload.per_page,
+        sortBy: action.payload.sort.by,
+        sortOrder: action.payload.sort.order,
+        total: action.payload.total,
+      });
+    case HW_MODEL_TABLE_FAILURE:
+      return state
+        .set('status', STATUS.ERROR)
+        .set('error', action.payload.error);
```

Comment:
Don't you want to set the `results` back to empty-array when error appear?

Reply:
If there is an error, the results are not going to be shown.
In that case the user can only refresh the page which clears the Redux state.
Although, I don't see any point in doing so, maybe it makes the code more readable.

Reply:
I see the state is our main source of truth and the place we go first when we want to understand/define what should actually get rendered.
I believe it must make sense so we can understand the application flow.

Reply:
Yep, I see your point here :dark_sunglasses: 
I updated the code.